src/include/stir/listmode/SPECTListModeData.h

```diff
@@ -0,0 +1,260 @@
+#endif
+
+START_NAMESPACE_STIR
+
+class Succeeded;
+class ExamInfo;
+
+/*!
+  \brief The base class for reading list mode data.
+  \ingroup listmode
+
+  \par What is list mode data?
+
+  List mode data is a format for storing detected counts as a list, as
+  opposed to a histogram. For each count, a list mode event contains
+  all the properties that the scanner can give you. Hence, the
+  list mode data gives you all the information about your acquisition
+  that you can possibly get. Which information this is obviously depends
+  on the scanner.
+
+  For most (all?) scanners, events are stored in chronological order.
+  In addition to events, time flags are inserted into the list mode data.
+  So, generally list mode data is a list of 'records', which can be
+  of different types. In STIR, this concept of a 'record' corresponds to
+  the ListRecord class and its relatives (see the documentation for
+  ListRecord).
+
+  \par Usage
+
+  For most applications, i.e. when one just wants to go through the list of
+  all events, the code would be as follows:
+
+  \code
+  shared_ptr<SPECTListModeData>
+    lm_data_sptr(read_from_file<SPECTListModeData>(filename));
+
+  // get a pointer to a 'template' record that will work for the scanner
+  // from which we're reading data
+  shared_ptr <SPECTListRecord> record_sptr =
+    lm_data_ptr->get_empty_record_sptr();
+  // give the record a simple name to avoid cluttering the code below
+  SPECTListRecord& record = *record_sptr;
+
+  double current_time = 0;
+  while (lm_data_sptr->get_next_record(record) == Succeeded::yes)
+    {
+      if (record.is_time())
+      {
+    current_time= record.time().get_time_in_secs();
+      }
+      if (record.is_event())
+      {
+        if (record.event().is_prompt())
+    { // do something
+        }
+        // ...
+      }
+    }
+  \endcode
+
+  In addition, there is a facility to 'remember' positions in the list,
+  and go back to one of these positions. This could be useful to
+  mark time frames. This goes as follows.
+
+  \code
+  // somehow I found out this is the start of the frame, so save it
+  SPECTListModeData::SavedPosition start_of_this_frame =
+     lm_data_sptr->save_get_position();
+
+  // now do something with this frame
+
+  // now get back to the start of the frame
+  if ( lm_data_sptr->set_get_position(start_of_this_frame)
+     != Succeeded::yes)
+    error("Help!");
+  \endcode
+
+  This class (and SPECTListRecord) is specific to SPET, i.e. to
+  single gamma detection. However, the only difference with the PET part
+  are the functions related to prompts and delayeds.
+  SPECT is handled by calling all single photon events 'prompts'.
+
+  \par Notes for developers
+
+  If you want to add a new type of list mode data, you have to make corresponding
+  derived classes of SPECTListModeData, SPECTListRecord etc. You also have to modify
+  make sure that read_from_file<SPECTListModeData> recognises your data. This
```

Comment:
This seems incorrect. we're using `read_from_file<ListModeData>` (see also comment about registries). I would cut this sentence, or in fact the whole paragraph (from L124)